### PR TITLE
parametric(python): re-enable previously failing tests

### PR DESCRIPTION
**Pull request by mabdinur**
## Motivation

Opentelemetry-api v1.26.0 was recently released: https://pypi.org/project/opentelemetry-api/#history. This broke ddtrace-opentelemetry support. 

This in compatibility was resolved in ddtrace v2.10.1.

This PR re-enables Otel tests

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

**Review by zacharycmontoya — APPROVED** [review]
LGTM, thanks for a quick fix that unblocks our tests

**Review by khanayan123 — APPROVED** [review]

File: utils/build/docker/python/install_ddtrace.sh [hunk 1]
```diff
@@ -22,6 +22,10 @@ else
     exit 1
 fi
 
+# Avoids incompatiblity with v1.26.0
+# FIX: https://github.com/DataDog/dd-trace-py/pull/9941
+pip install "opentelemetry-api<1.25.0"
+
```
Comment (mabdinur) [inline utils/build/docker/python/install_ddtrace.sh:28]:
```suggestion
```

**Review by mabdinur — COMMENTED** [review]
Fix was released

**Review by cbeauchesne — CHANGES_REQUESTED** [review]
CI is happy -> I'm happy :) 

Though, there is a conflict to resolve.

File: tests/parametric/test_otel_span_methods.py [hunk 2]
```diff
@@ -414,7 +411,10 @@ def test_otel_set_span_status_ok(self, test_agent, test_library):
         assert span.get("resource") == "ok_span"
 
     @bug(context.library < "ruby@2.2.0", reason="Older versions do not generate datadog spans with the correct ids")
-    @bug(context.library >= "python@2.9.3", reason="APMAPI-180")
+    @flaky(
+        context.library == "dotnet",
+        reason="Flaky due to span order, tests should never use traces[0] when there are multiple spans",
+    )
```
Comment (mabdinur) [inline tests/parametric/test_otel_span_methods.py:417]:
```suggestion
```

**Review by cbeauchesne — APPROVED** [review]

**Pull request merged**